### PR TITLE
Server post

### PR DESCRIPTION
**Pull request by halimahexe**
Relates #4

Created an app.post function for users to submit new questions. I had to add `express.json()` body parser to parse the `req.body` and the console.log works. However, the writeData function isn't working so it's not actually updating the data.json with the new information. This needs to be looked at.

### CODE REVIEW

**Comment by halimahexe** [comment]
I fixed the writeData function. `jsonDB.push` wasn't working because it was an object, not an array so I just had to target `jsonDB.questions.push` instead and it worked!

File: src/index.ts [hunk 1]
```diff
@@ -7,14 +7,29 @@ const library = 'data.json';
...
 
+// request body template
+type Question = {
+	id?: number;
+	category: string;
+	difficulty: 'easy' | 'medium' | 'hard';
+	question: string;
+	options: [string, string, string, string];
+	answer: string;
+	favourited?: boolean;
```
Comment (that-Gui) [inline src/index.ts:24]: remove ? and replace with boolean || null

**Review by Gaajia — APPROVED** [review]

File: src/index.ts [hunk 2]
```diff
@@ -7,14 +7,29 @@ const library = 'data.json';
...
+	timestamp?: string;
```
Comment (that-Gui) [inline src/index.ts:25]: remove ? and replace with string || null

**Review by that-Gui — CHANGES_REQUESTED** [review]
can you remove the ?  and use || instead plox
just makes the code more readable

**Comment by halimahexe** [comment]
> can you remove the ? and use || instead plox just makes the code more readable

I think that would be doing something slightly different, where it would make it necessary to have all of those properties in the objects, rather than what `?` is doing. `?` means the property is optional so if the type object doesn't have that property, it won't come up with an error.

**Review by that-Gui — APPROVED** [review]
LGTM

**Comment by halimahexe** [comment]
Closes #4

**Pull request merged**